### ISSUE TITLE
Signing function unit test

### ISSUE THREAD

**Issue by hanwencheng** [paritytech/parity-signer#387]
Thanks for the fix for #382 , yesterday I debugged to the [line](https://github.com/paritytech/parity-signer/pull/383/files#diff-bcf08b8f1f5d5966fe583d09d70a07ecR88) of `keccak(asString(data))` But not one step deeper, since it start involving the encoding functions which I am not familiar with, and it will need to record some test data for it, so I just go back to other unsolved issues.

Besides, which I noticed from this bug is: for most of the PRs, I do not go through the whole signing process to test but only test it until the step of scanning the QR code. 

The result is, firstly the problem takes weeks to be discovered, secondly it will take time to target the error with commits afterwards.

I think it would be better to set up a unit test suit with sample data for both message and transaction signing on Ethereum and Substrate.

A possible structure could be:

```
describe('ethereum data parsing and signing', () => {
  it('should correctly parse the transaction data', () => {

  });

  it('should correctly sign the transaction data', () => {

  });

  it('should correctly parse the message data', () => {

  });

  it('should correctly sign the message data', () => {

  })
});

describe('substrate data parsing and signing', () => {
  it('should correctly parse the transaction data', () => {

  });

  it('should correctly sign the transaction data', () => {

  });

  it('should correctly parse the message data', () => {

  });

  it('should correctly sign the message data', () => {

  })
});
```

**Comment by yjkimjunior** [comment]
100%. This was brought up before as well, the only thing is that the signing, hashing, etc. are done outside of the JS environment so we won't be able to "unit test" a la Jest per say. Parsing we can do.

At the moment we unit test on either side of the bridge and call it a day, but as we've seen this leaves a lot to be desired.

Which brings me to a bigger issue which is the need for e2e integration tests. We'd need a suite of tests for both the Android and iOS bridges. I haven't done this before but https://medium.com/@SkyscannerEng/testing-react-native-ios-bridges-80c730659a83 seems like a promising start.

**Comment by hanwencheng** [comment]
closed by #549 .
